src/pages/allSearchResults/index.jsx

```diff
@@ -0,0 +1,119 @@
+import ProfileCircle from '../../components/profileCircle';
+import Header from '../../components/header';
+import Navigation from '../../components/navigation';
+import Card from '../../components/card';
+import { getUsers } from '../../service/apiClient';
+import './style.css';
+
+const AllSearchResults = () => {
+  const location = useLocation();
+  const { results: initialResults, searchVal: initialSearchVal } = location.state || { results: [], searchVal: '' };
+  const [searchVal, setSearchVal] = useState(initialSearchVal);
+  const [results, setResults] = useState(initialResults);
+  const [cohorts, setCohorts] = useState([]);    
+
+
+  useEffect(() => {
+      getUsers().then(setCohorts);
+      window.scrollTo(0, 0);
+  }, []);
+
+  const onChange = (e) => {
+    setSearchVal(e.target.value);
+    const filteredResults = cohorts.filter((cohort) => {
+      if (cohort.firstName && cohort.lastName) {
+        const fullName = `${cohort.firstName || ''} ${cohort.lastName || ''}`.toLowerCase();
+        return fullName.includes(e.target.value.toLowerCase());
+      }
+      return false;
+    });
+    setResults(filteredResults);
+  };
+
+  const onClickStudent = (id) => {
+    setSelectedProfileId(id);
+    setIsStudentModalVisible(true);
+  };
+
+
+  return (
+      <>
+        <div className='container'>              
+            <Header />
+            <Navigation className="left-bar" />
+            <main className='search-results-container'>
+                <div>
+                    <div className='top'>
+                          <div className='title'>
+                              <Link to='/'>
+                                <ArrowLeftIcon />              
+                              </Link>
+                            <h2>Search results</h2>      
+                        </div>
+                        <div className='search-bar-container'>
+                            <Card name="search-bar">
+                                <div className='search-page-search-bar'>
+                                    <SearchIcon />
+                                    <input className="search-page-input"
+                                    type="search"
+                                    name="Search"
+                                    value={searchVal}
+                                    onChange={onChange}
+                                    placeholder="Search for people"
+                                    />
+                                </div>
+                            </Card>                              
+                        </div>
+                    </div>
+
+                    <div className='search-results'>
+                          <Card className='search-results-card' name="results">
+                            {results.length === 0 ? (
+                                <p>No results found.</p>
+                            ) : (
+                                <ul className='search-results-list'>
```

Comment:
```jsx
{results.length === 0 && <p>No results found.</p>}

{results.length !== 0 &&
  <ul> ...
}
```

Always try to reach for react's conditional rendering rather than ternary's